tests/core/pyspec/eth2spec/test/capella/epoch_processing/test_process_full_withdrawals.py

```diff
@@ -1,39 +1,40 @@
+from random import Random
+
 from eth2spec.test.context import (
     with_capella_and_later,
     spec_state_test,
 )
-from eth2spec.test.helpers.epoch_processing import run_epoch_processing_with
-
-
-def set_validator_withdrawable(spec, state, index, withdrawable_epoch=None):
-    if withdrawable_epoch is None:
-        withdrawable_epoch = spec.get_current_epoch(state)
-
-    validator = state.validators[index]
-    validator.withdrawable_epoch = withdrawable_epoch
-    validator.withdrawal_credentials = spec.ETH1_ADDRESS_WITHDRAWAL_PREFIX + validator.withdrawal_credentials[1:]
-
-    assert spec.is_fully_withdrawable_validator(validator, withdrawable_epoch)
+from eth2spec.test.helpers.random import (
+    randomize_state,
+)
+from eth2spec.test.helpers.epoch_processing import (
+    run_epoch_processing_to,
+)
+from eth2spec.test.helpers.withdrawals import (
+    set_validator_fully_withdrawable,
+)
 
 
 def run_process_full_withdrawals(spec, state, num_expected_withdrawals=None):
+    run_epoch_processing_to(spec, state, 'process_full_withdrawals')
+
     pre_next_withdrawal_index = state.next_withdrawal_index
     pre_withdrawal_queue = state.withdrawal_queue.copy()
     to_be_withdrawn_indices = [
         index for index, validator in enumerate(state.validators)
-        if spec.is_fully_withdrawable_validator(validator, spec.get_current_epoch(state))
+        if spec.is_fully_withdrawable_validator(validator, state.balances[index], spec.get_current_epoch(state))
     ]
 
     if num_expected_withdrawals is not None:
         assert len(to_be_withdrawn_indices) == num_expected_withdrawals
     else:
         num_expected_withdrawals = len(to_be_withdrawn_indices)
 
-    yield from run_epoch_processing_with(spec, state, 'process_full_withdrawals')
+    yield 'pre', state
+    spec.process_full_withdrawals(state)
+    yield 'post', state
```

Comment:
any reason to not use the `run_epoch_processing_with` helper?

Reply:
yes this pattern is useful when you need to capture information right before the function is called.

In this case, balance updates can/will occur during those interim epoch processings. So in some cases this changes the `to_be_withdrawn` indices in some test cases